### PR TITLE
Fail deploy helpfully when unmanaged pod preexists

### LINKED ISSUE #108: pod.rb:82:in `block in fetch_logs': undefined method `to_datetime' for nil:NilClass
Got this exception while trying to deploy `wedge-viewer` for the first time, looks like the `@deploy_started` variable hasn't been set by the time we try to use it there :(

Full trace:
```

[INFO][2017-06-06 23:07:21 +0000]	----------------------------Phase 2: Checking initial resource statuses-----------------------------
[INFO][2017-06-06 23:07:23 +0000]	
[INFO][2017-06-06 23:07:23 +0000]	------------------------------------------Result: FAILURE-------------------------------------------
[FATAL][2017-06-06 23:07:23 +0000]	No actions taken
/usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/kubernetes_resource/pod.rb:82:in `block in fetch_logs': undefined method `to_datetime' for nil:NilClass (NoMethodError)
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/kubernetes_resource/pod.rb:77:in `each'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/kubernetes_resource/pod.rb:77:in `each_with_object'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/kubernetes_resource/pod.rb:77:in `fetch_logs'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/kubernetes_resource/pod.rb:98:in `display_logs'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/kubernetes_resource/pod.rb:26:in `sync'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/runner.rb:90:in `each'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/lib/kubernetes-deploy/runner.rb:90:in `run'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/lib/ruby/gems/2.3.0/gems/kubernetes-deploy-0.7.4/exe/kubernetes-deploy:61:in `<top (required)>'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/bin/kubernetes-deploy:22:in `load'
	from /usr/lib/ruby-shopify/ruby-shopify-2.3.3/bin/kubernetes-deploy:22:in `<main>'
```
Deploy here: https://shipit.shopify.io/shopify/wedge-viewer/production/deploys/452708

The first deploy for this stack failed in an interesting way where I forced shipit to deploy before the container was ready (because I forgot there was a container build step), so maybe it's failure condition has something to do with why this is happening now? See it here: https://shipit.shopify.io/shopify/wedge-viewer/production/deploys/452703

### PR DESCRIPTION
**Pull request by KnVerey**
Replaces https://github.com/Shopify/kubernetes-deploy/pull/109 because that file changed a bunch since I opened it.
Fixes #108 

image

### CODE REVIEW

**Comment by KnVerey** [comment]
Ping @ibawt @stefanmb could one of you please have a quick look at this? It isn't all that critical, but it has been hanging around for a while so I'd like to get it in. 😄

File: lib/kubernetes-deploy/kubernetes_resource/pod.rb [hunk 1]
```diff
@@ -19,6 +19,7 @@ def sync(pod_data = nil)
       if pod_data.blank?
         raw_json, _err, st = kubectl.run("get", type, @name, "-a", "--output=json")
         pod_data = JSON.parse(raw_json) if st.success?
+        raise_predates_deploy_error if pod_data.present? && unmanaged? && !@deploy_started
```
Comment (stefanmb) [inline lib/kubernetes-deploy/kubernetes_resource/pod.rb]: > An object is present if it's not blank.

> An object is blank if it's false, empty, or a whitespace string.

How can `pod_data.present?` be true when `pod_data.blank?` is true?

Comment (stefanmb) [inline lib/kubernetes-deploy/kubernetes_resource/pod.rb]: > An object is present if it's not blank.

> An object is blank if it's false, empty, or a whitespace string.

How can `pod_data.present?` be true when `pod_data.blank?` is true?

Comment (stefanmb) [inline lib/kubernetes-deploy/kubernetes_resource/pod.rb]: Why `!@deploy_started`?

Comment (ibawt) [inline lib/kubernetes-deploy/kubernetes_resource/pod.rb]: should we rename pod_data to prevent the confusion?

Comment (ibawt) [inline lib/kubernetes-deploy/kubernetes_resource/pod.rb]: deploy_started seems like state is leaking from the deploy into the resources

Comment (KnVerey) [inline lib/kubernetes-deploy/kubernetes_resource/pod.rb]: As discussed IRL, `deploy_started` should probably have been named `deploy_started_at`: it is a timestamp indicating when _this specific resource_ was deployed (or it is nil when it hasn't been deployed yet). What we're doing here is checking whether an unmanaged pod existed before the current deploy created it.

**Pull request merged**